src/views/Visualizer/View/View.js

```diff
@@ -345,16 +345,20 @@ export class Visualizer extends React.Component {
     const filters = filterLayersStatesFromLayersState(layersTreeState)
       .filter(([{ filters: { layer } = {} }, { filters: layerFilters }]) =>
         layer && (query || layerFilters))
-      .map(([layer, { filters: properties = {} }]) => ({
-        layer,
-        properties: {
-          ...Object.keys(properties).reduce((all, key) => ({
-            ...all,
-            ...getSearchParamFromProperty(properties, layer.filters.form, key),
-          }), {}),
-        },
-        index: layer.filters.layer,
-      }));
+      .map(([layer, { filters: properties = {} }]) => {
+        const { filters: { form, layer: index } } = layer;
+        return {
+          layer,
+          properties: {
+            ...Object.keys(properties).reduce((all, key) => ({
+              ...all,
+              ...(form.some(({ property }) => property ===  key) && getSearchParamFromProperty(properties, form, key)),
```

Comment:
As seen together over Jitsi, this could maybe been improved (regarding readability)